caffe2/operators/variable_length_sequence_padding.h

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VARIABLE_LENGTH_SEQUENCE_PADDING_H
+#define VARIABLE_LENGTH_SEQUENCE_PADDING_H
+
+#include "caffe2/core/context.h"
+#include "caffe2/core/operator.h"
+#include "caffe2/utils/math.h"
+
+namespace caffe2 {
+namespace detail {
+
+template <typename T, typename Context>
+void VariableLengthSequencePadding(
+    int N,
+    int B,
+    int M,
+    T* X,
+    const int32_t* seqLengths,
+    const T padValue,
+    Context* /*context*/) {
+  for (int i = 0; i < N; i++) {
```

Comment:
I think this can be easily vectorized using Eigen. Might be faster to do it right away instead of doing a slow operator and then getting feedbacks that onnx->caffe2 models are slow, digging down into why and fixing things. In some cases IMO we can just write more optimal code when it is not that much effort.

Reply:
sure, can you point me to an existing example I can mimic? I'm not familiar with Eigen

Reply:
You could search for Eigne usage in Caffe2 repo :)

https://github.com/caffe2/caffe2/blob/01827c153db96349745a544148c1ff0386c5ef9e/caffe2/operators/elementwise_div_op.cc

Reply:
There are lot of examples in the repo, just search for Eigen. A more relevant to RNN states dropping is in D5051296. It was reverted as doing it inside an RNN was prone to alignment issues. But as you do it outside, it is fine.